.github/workflows/build.yaml

```diff
@@ -4,44 +4,24 @@ on:
   pull_request:
   push:
     branches-ignore:
-      - 'update-*'
+      - "update-*"
   workflow_dispatch:
 
 jobs:
-  x86_64-linux:
-    runs-on: ubuntu-latest
-
-    steps:
-      - name: Install Nix
-        uses: DeterminateSystems/nix-installer-action@main
-
-      - name: Install Magic Nix Cache
-        uses: DeterminateSystems/magic-nix-cache-action@main
-
-      - run: nix build github:$GITHUB_REPOSITORY/$GITHUB_SHA --accept-flake-config --json | nix run nixpkgs#jq -- -r '.[].outputs | to_entries[].value' | nix run nixpkgs#cachix push toyvo
-        env:
-          CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
-
-  aarch64-linux:
-    runs-on: [self-hosted, linux, ARM64]
-
-    steps:
-      - run: nix build github:$GITHUB_REPOSITORY/$GITHUB_SHA --accept-flake-config --json | nix run nixpkgs#jq -- -r '.[].outputs | to_entries[].value' | nix run nixpkgs#cachix push toyvo
-        env:
-          CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
-
-  aarch64-darwin:
-    runs-on: [self-hosted, macOS, ARM64]
-
-    steps:
-      - run: nix build github:$GITHUB_REPOSITORY/$GITHUB_SHA --accept-flake-config --json | nix run nixpkgs#jq -- -r '.[].outputs | to_entries[].value' | nix run nixpkgs#cachix push toyvo
-        env:
-          CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
-
-  x86_64-darwin:
-    runs-on: [self-hosted, macOS, x64]
-
+  build:
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      matrix:
+        runner:
+          - ubuntu-latest
+          - [self-hosted, linux, ARM64]
+          - [self-hosted, macOS, ARM64]
+          - [self-hosted, macOS, x64]
     steps:
-      - run: nix build github:$GITHUB_REPOSITORY/$GITHUB_SHA --accept-flake-config --json | nix run nixpkgs#jq -- -r '.[].outputs | to_entries[].value' | nix run nixpkgs#cachix push toyvo
-        env:
-          CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
+      - uses: actions/checkout@v4
+      - uses: cachix/install-nix-action@v25
+      - uses: cachix/cachix-action@v14
+        with:
+          name: toyvo
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
```

Comment:
I appreciate the simplification of the build file, I had to update my local runners a little bit but its building successfully there too now. But this setting of the auth token isn't working. I'll try going back to the environment variable and see that works https://github.com/ToyVo/nh_darwin/actions/runs/11592022500/job/32281734435#step:9:3

Reply:
I simply had to remove the quotes and pushing to cachix works